core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/StopStartDownscaleDecommissionViaCMHandler.java

```diff
@@ -88,10 +89,13 @@ protected Selectable doAccept(HandlerEvent<StopStartDownscaleDecommissionViaCMRe
             Set<String> hostNames = getHostNamesForPrivateIds(request.getInstanceIdsToDecommission(), stack);
             LOGGER.debug("Attempting to decommission hosts. count={}, hostnames={}", hostNames.size(), hostNames);
 
-            HostGroup hostGroup = hostGroupService.getByClusterIdAndName(cluster.getId(), request.getHostGroupName())
-                    .orElseThrow(NotFoundException.notFound("hostgroup", request.getHostGroupName()));
+            Map<String, InstanceMetaData> hostsToRemove = new HashMap<>();
+            for (String hostGroupName : request.getHostGroupNames()) {
```

Comment:
This is specifically not supposed to handle multiple hostGroups (at least for some time). Better to throw an error if there's more than one.